torchaudio/models/_wavernn.py

```diff
@@ -99,9 +99,129 @@ def forward(self, x: Tensor) -> Tensor:
 
         Shape:
             - x: :math:`(N, S, T)`.
-            - output: :math:`(N, P, T-2*pad)`.
+            - output: :math:`(N, P, T - 2 * pad)`.
         where N is the batch size, S is the number of input sequence,
-        P is the number of ouput sequence, T is the length of input sequence.
+        P is the number of output sequence, T is the length of input sequence.
         """
 
         return self.melresnet_model(x)
+
+
+class _Stretch2d(nn.Module):
+    r"""This is a two-dimensional stretch layer. It is a block used in WaveRNN. WaveRNN is based on the paper
+    "Efficient Neural Audio Synthesis". Nal Kalchbrenner, Erich Elsen, Karen Simonyan, Seb Noury,
+    Norman Casagrande, Edward Lockhart, Florian Stimberg, Aaron van den Oord, Sander Dieleman,
+    Koray Kavukcuoglu. arXiv:1802.08435, 2018.
+
+    Args:
+        x_scale: the scale factor in x axis (required).
+        y_scale: the scale factor in y axis (required).
+
+    Examples::
+        >>> stretch2d = _Stretch2d(x_scale=1, y_scale=1)
+
+        >>> input = torch.rand(10, 1, 100, 512)
+        >>> output = stretch2d(input)
+    """
+
+    def __init__(self,
+                 x_scale: int,
+                 y_scale: int) -> None:
+        super().__init__()
+
+        self.x_scale = x_scale
+        self.y_scale = y_scale
+
+    def forward(self, x: Tensor) -> Tensor:
+        r"""Pass the input through the _Stretch2d layer.
+
+        Args:
+            x: the input sequence to the _Stretch2d layer (required).
+
+        Shape:
+            - x: :math:`(N, C, S, T)`.
+            - output: :math:`(N, C, S * y_scale, T * x_scale)`.
+        where N is the batch size, C is the channel size, S is the number of input sequence,
+        T is the length of input sequence.
+        """
+
+        n, c, s, t = x.size()
```

Comment:
Expecting a four tuple is a little rigid, isn't it?

Reply:
e.g. some functions support `...` to mean an arbitrary number of dimensions, see functionals in torchaudio.

Reply:
I have updated it.